samples/matter/common/src/diagnostic/diagnostic_logs_provider.h

```diff
@@ -22,6 +22,11 @@ namespace Nrf::Matter
 
 class DiagnosticLogProvider : public chip::app::Clusters::DiagnosticLogs::DiagnosticLogsProviderDelegate {
 public:
+	struct IntentData {
+		chip::app::Clusters::DiagnosticLogs::IntentEnum mIntent;
+		void *mIntentImpl = nullptr;
```

Comment:
I guess it is better to use here an abstract class and then define 3 intent implementations based on that rather than using `void *`. `Void*` can suggest that it can be used for different purposes, whereas you need to use a specific API inside DiagnosticLogProvider methods.

Reply:
Done, I've implemented abstract DiagnosticLogsIntentImpl class. Please re-review.